www/class/CentreonLDAPAdmin.class.php

```diff
@@ -99,19 +99,44 @@ public function getLdapParameters()
         return $tab;
     }
 
+    /**
+     * used to sanitize key and value in array
+     * @param array<mixed> $inputArray
+     * @return array<mixed>
+     */
+    public function sanitizeInputArray(array $inputArray): array
+    {
+        $sanitizedArray = [];
+        foreach ($inputArray as $key => $value) {
+            if ($key === 'address') {
+                $key = filter_var($key, FILTER_SANITIZE_STRING);
+                $value = filter_var($value, FILTER_SANITIZE_STRING);
+            } else {
+                $key = filter_var($key, FILTER_VALIDATE_INT);
+                $value = filter_var($value, FILTER_VALIDATE_INT);
+            }
+            if (false !== $key && false !== $value) {
+                $sanitizedArray[$key] = $value;
+            }
+        }
+        return $sanitizedArray;
+    }
```

Comment:
To avoid to duplicate this code (in ldap.php, etc), could you move it in "CentreonUtils" for example below "escapeSecure".

Reply:
done